homeassistant/helpers/service.py

```diff
@@ -189,7 +190,22 @@ def async_prepare_call_from_config(
 
     domain, service = domain_service.split(".", 1)
 
-    target = config.get(CONF_TARGET)
+    target = {}
+    if CONF_TARGET in config:
+        conf = config.get(CONF_TARGET)
+        try:
+            template.attach(hass, conf)
+            target.update(template.render_complex(conf, variables))
+            if CONF_ENTITY_ID in target:
+                target[CONF_ENTITY_ID] = cv.comp_entity_ids(target[CONF_ENTITY_ID])
```

Comment:
We don't need this at all I think. It should be validated when the service is called. Can also drop catching vol.Invalid

Reply:
Without & invalid result:
![image](https://user-images.githubusercontent.com/195327/108930267-ea42cf00-7645-11eb-975e-0f1527ef61f4.png)

With the existing (additional) validation:

![image](https://user-images.githubusercontent.com/195327/108930292-f29b0a00-7645-11eb-9ca9-184c396e2a12.png)

The first isn't really helpful